packages/destination-actions/src/destinations/launchdarkly-audiences/syncAudience/index.ts

```diff
@@ -0,0 +1,74 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import { processPayload } from '../../launchdarkly-audiences/syncAudience/custom-audience-operations'
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Sync Audience',
+  description: 'Sync Engage Audiences to LaunchDarkly segments',
+  defaultSubscription: 'type = "identify" or type = "track"',
+  fields: {
+    custom_audience_name: {
+      label: 'Custom Audience Name',
+      description: 'Name of custom audience list to which user identifier should added/removed',
+      type: 'string',
+      required: true,
+      default: {
+        '@path': '$.context.personas.computation_key'
+      }
+    },
+    segment_computation_action: {
+      label: 'Segment Computation Action',
+      description: "Segment computation class used to determine if action is an 'Engage-Audience'",
+      type: 'string',
+      required: true,
+      default: {
+        '@path': '$.context.personas.computation_class'
+      }
+    },
+    context_kind: {
+      label: 'Context kind',
+      description: 'LaunchDarkly context kind',
+      type: 'string',
+      required: true,
+      default: 'user',
+      choices: [{ label: 'User', value: 'user' }]
+    },
+    context_key: {
+      label: 'Context key',
+      description: 'LaunchDarkly context key',
+      type: 'string',
+      required: true,
+      default: {
+        '@if': {
+          exists: { '@path': '$.userId' },
+          then: { '@path': '$.userId' },
+          else: { '@path': '$.email' }
+        }
+      }
+    },
+    traits_or_props: {
```

Comment:
Again, I'm not sure if we want users to be able to configure this because it will probably add confusion and a chance for them to break something.

Reply:
Similar to my response for `segment_computation_action` field, it doesnt seem we have a way to access this prop directly from the raw payload in the `perform` functions. Looks like we have to declare it as a field and use the mapper to map the value from the raw payload into the transformed field.